open-apparel-registry/open-apparel-registry
# Update aliases when deleting a facility

## Overview

Make required updates to existing FacilityAlias records to allow facilities with aliases to be deleted.

Connects #597 	

## Notes

Fixes an edge case that was missed when implementing #616 

## Testing Instructions

* Reset the DB
  * `./scripts/manage resetdb`
  * `./scripts/manage loadfixtures`
  * `./scripts/manage processfixtures`
* Log in as `c1@example.com` and generate an API key on the profile page. Save this for later.
* On the map page, search for `Regina Miracle Intimate`. Verify that the facility has three contributors.
* Copy the OAR ID and submit a delete API request 
```
curl --request DELETE \
  --url http://localhost:8081/api/facilities/{{OAR ID}} \
  --header 'authorization: Token {{API KEY}}'
```
* Search for `Regina Miracle Intimate` and verify that the facility now has only 2 contributors.
* Copy the new OAR ID and submit another delete API request 
```
curl --request DELETE \
  --url http://localhost:8081/api/facilities/{{NEW OAR ID}} \
  --header 'authorization: Token {{API KEY}}'
```
* Verify that this succeeds
* Search for `Regina Miracle Intimate` and verify that the facility now has only 1 contributor.
* Copy the new OAR ID and submit another delete API request 
```
curl --request DELETE \
  --url http://localhost:8081/api/facilities/{{LAST OAR ID}} \
  --header 'authorization: Token {{API KEY}}'
```
* Verify that this succeeds

## Checklist
